kategory-docs/docs/datatypes/either/README.md

```diff
@@ -0,0 +1,213 @@
+kind of exceptions (if any) a function may throw, we have to dig through the source code. 
+Then to handle these exceptions, we have to make sure we catch them at the call site. This 
+all becomes even more unwieldy when we try to compose exception-throwing procedures.
+
+```kotlin:ank
+import kategory.*
+
+val throwsSomeStuff: (Int) -> Double = {x -> x.toDouble()}
+val throwsOtherThings: (Double) -> String = {x -> x.toString()}
+val moreThrowing: (String) -> List<String> = {x -> listOf(x) }
+val magic = throwsSomeStuff.andThen(throwsOtherThings).andThen(moreThrowing)
+magic
+```
+
+Assume we happily throw exceptions in our code. Looking at the types, any of those functions can 
+throw any number of exceptions, we don't know. When we compose, exceptions from any of the constituent
+functions can be thrown. Moreover, they may throw the same kind of exception 
+(e.g. `IllegalArgumentException`) and thus it gets tricky tracking exactly where that exception came from.
+
+How then do we communicate an error? By making it explicit in the data type we return.
+
+## EITHER VS VALIDATED
+
+In general, `Validated` is used to accumulate errors, while `Either` is used to short-circuit a computation 
+upon the first error. For more information, see the `Validated` vs `Either` section of the `Validated` documentation.
+
+More often than not we want to just bias towards one side and call it a day - by convention,
+the right side is most often chosen.
+
+```kotlin:ank
+val right: Either<String, Int> = Either.Right(5)
+right
+``` 
+
+```kotlin:ank
+val left: Either<String, Int> = Either.Left("Something went wrong")
+left
+``` 
+Because `Either` is right-biased, it is possible to define a Monad instance for it.
+
+Since we only ever want the computation to continue in the case of `Right` (as captured by the right-bias nature), 
+we fix the left type parameter and leave the right one free.
+
+So the flatMap method is right-biased:
+
+```kotlin:ank
+val right: Either<String, Int> = Either.Right(5)
+right.flatMap{Either.Right(it + 1)}
+
+val left: Either<String, Int> = Either.Left("Something went wrong")
+left.flatMap{Either.Right(it + 1)}
+``` 
+
+## USING EITHER INSTEAD OF EXCEPTIONS
+
+As a running example, we will have a series of functions that will parse a string into an integer, 
+take the reciprocal, and then turn the reciprocal into a string.
+
+In exception-throwing code, we would have something like this:
+
+```kotlin:ank:silent
+object ExceptionStyle {
+    fun parse(s: String): Int =
+        if (s.matches(Regex("-?[0-9]+"))) s.toInt()
+        else throw NumberFormatException("$s is not a valid integer.")
+
+    fun reciprocal(i: Int): Double =
+        if (i == 0) throw IllegalArgumentException("Cannot take reciprocal of 0.")
+        else 1.0 / i
+
+    fun stringify(d: Double): String = d.toString()
+}
+``` 
+
+Instead, let's make the fact that some of our functions can fail explicit in the return type.
```

Comment:
Nit, I would say:

Instead, let's make explicit in the return type that some of our functions can fail: